### PR TITLE
[dbnode] Read only namespaces

### PR DESCRIPTION
**Pull request by linasm**
**What this PR does / why we need it**:
Adds support for read only namespaces.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
NONE

**Does this PR require updating code package or user-facing documentation?**:
NONE

### CODE REVIEW

File: src/dbnode/namespace/types.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ type SchemaRegistry interface {
 
 // Metadata represents namespace metadata information
 type Metadata interface {
-	// Equal returns true if the provide value is equal to this one
+	// Equal returns true if the provided value is equal to this one
```
Comment (arnikola) [inline src/dbnode/namespace/types.go:216]: nit; mind adding `.` to all of these including the existing ones?

File: src/dbnode/storage/namespace.go [hunk 2]
```diff
@@ -1256,7 +1266,7 @@ func (n *dbNamespace) ColdFlush(flushPersist persist.FlushPreparer) error {
 
 	// If repair is enabled we still need cold flush regardless of whether cold writes is
 	// enabled since repairs are dependent on the cold flushing logic.
-	if !n.nopts.ColdWritesEnabled() && !n.nopts.RepairEnabled() {
+	if n.ReadOnly() || !(n.nopts.ColdWritesEnabled() || n.nopts.RepairEnabled()) {
```
Comment (arnikola) [inline src/dbnode/storage/namespace.go:1269]: nit; may be clearer to read as the following so the logic is more explicit
```
repairEnabled := !n.nopts.ColdWritesEnabled() && !n.nopts.RepairEnabled()
if n.ReadOnly() || repairEnabled {
```

**Review by arnikola — APPROVED** [review]
lgtm /w nits

File: src/dbnode/namespace/types.go (hunk 1 above)
Comment (linasm) [reply src/dbnode/namespace/types.go:216]: 👍

File: src/dbnode/storage/namespace.go (hunk 2 above)
Comment (linasm) [reply src/dbnode/storage/namespace.go:1269]: Will do it, but in a bit different way.

**Pull request merged**